src/main/java/org/sonarqube/gradle/SonarQubePlugin.java

```diff
@@ -117,18 +119,26 @@ private SonarQubeTask createTask(final Project project, final Map<Project, Actio
       .filter(p -> isAndroidProject(p) && !p.getExtensions().getByType(SonarQubeExtension.class).isSkipProject())
       .map(p -> {
         BaseVariant variant = AndroidUtils.findVariant(p, p.getExtensions().getByType(SonarQubeExtension.class).getAndroidVariant());
-        try {
-          return p.getTasks().getByName("compile" + capitalize(variant.getName()) + "JavaWithJavac");
-        } catch (UnknownTaskException e) {
-          return null;
-        }
+        List<Task> allCompileTasks = new ArrayList<>();
+        addTaskByName(p, "compile" + capitalize(variant.getName()) + "JavaWithJavac", allCompileTasks);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "compile" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1192)